app/samples/phase_detection/src/main.cpp

```diff
@@ -0,0 +1,284 @@
+    constexpr double boost_threshold_mps2 = 50;
+
+    // < 3.5G-ish for a quarter second
+    // weird since this example is doing what grim did and doing it based on magnitude not a specific axis
+    // Deceleration is still acceleration which is why we cant just say < 1G
+    constexpr uint32_t coast_time_ms = 250;
+    constexpr double coast_threshold_mps2 = 35;
+    CDebouncer<ThresholdDirection::Over, double> boost_detector{boost_time_ms, boost_threshold_mps2};
+    CDebouncer<ThresholdDirection::Under, double> coast_detector{coast_time_ms, coast_threshold_mps2};
+
+    CAccelerometer acc(*DEVICE_DT_GET_ONE(openrocket_imu));
+    if (!acc.IsReady()) {
+        LOG_WRN("Accelerometer not ready");
+    }
+    controller.SubmitEvent(Sources::IMU1, Events::PadReady);
+    controller.WaitUntilEvent(Events::PadReady);
+
+    while (!controller.HasEventOccured(Events::GroundHit)) {
+        k_timer_status_sync(&imu_timer);
+        uint32_t timestamp = k_uptime_get();
+
+        if (!acc.UpdateSensorValue()) {
+            LOG_ERR("Failure reading imu");
+        }
+        double x = acc.GetSensorValueDouble(SENSOR_CHAN_ACCEL_X);
+        double y = acc.GetSensorValueDouble(SENSOR_CHAN_ACCEL_Y);
+        double z = acc.GetSensorValueDouble(SENSOR_CHAN_ACCEL_Z);
+        double mag = sqrt((x * x) + (y * y) + (z * z));
+
+        // Boost Detecting
+        if (!controller.HasEventOccured(Events::Boost)) {
+            boost_detector.feed(timestamp, mag);
+            coast_detector.feed(timestamp, mag);
+            if (boost_detector.passed()) {
+                controller.SubmitEvent(Sources::IMU1, Events::Boost);
+            }
+        } else if (!controller.HasEventOccured(Events::Coast)) {
+            coast_detector.feed(timestamp, mag);
+            if (coast_detector.passed()) {
+                controller.SubmitEvent(Sources::IMU1, Events::Coast);
+            }
+        }
+    }
+}
+// K_THREAD_DEFINE(imu_thread, 1024, imu_thread_f, NULL, NULL, NULL, 0, 0, 0);
+
+using SampleType = LinearFitSample<double>;
+static constexpr std::size_t window_size = 10;
+using SummerType = CRollingSum<SampleType, window_size>;
+
+struct Line {
+    double m;
+    double b;
+    constexpr bool operator==(const Line &rhs) const { return m == rhs.m && b == rhs.b; }
+};
+// Value returned when we don't have enough information to fit a line
+constexpr Line bad_line{0, 0};
+
+Line find_line(const SummerType &summer) {
+    std::size_t N = summer.Size();
+    SampleType E = summer.Sum();
+    double denom = (N * E.xx - E.x * E.x);
+    if (denom == 0) {
+        // printf("Would have divided by 0\n");
+        return bad_line;
+    }
+    double m = (N * E.xy - E.x * E.y) / denom;
+    double b = (E.y - m * E.x) / N;
+    return Line{m, b};
+}
+
+double rrc3_altitude_conversion_to_feet(double P_sta_kpa) {
+    double P_sta_mbar = P_sta_kpa * 10.0;
+    double alt_ft = (1 - pow(P_sta_mbar / 1013.25, 0.190284)) * 145366.45;
```

Comment:
Also can we make these constants and not magic numbers

Reply:
uh, prolly. i need to go study the standard atmospheere to figure out what they are lol

Reply:
Thats why I asked lol